File: job-application-automation/src/detectAts.ts
import type { AtsPlatform } from './types.js';

export function detectAts(jobUrl: string): AtsPlatform {
  const host = new URL(jobUrl).hostname;

  if (host.includes('greenhouse.io') || host.includes('grnh.se')) {
    return 'greenhouse';
  }
  if (host.includes('lever.co')) {
    return 'lever';
  }
  if (host.includes('ashbyhq.com')) {
    return 'ashby';
  }
  return 'unknown';
}

File: job-application-automation-py/detect_ats.py
from urllib.parse import urlparse

from types_ import AtsPlatform


def detect_ats(job_url: str) -> AtsPlatform:
    host = urlparse(job_url).hostname or ""

    if "greenhouse.io" in host or "grnh.se" in host:
        return "greenhouse"
    if "lever.co" in host:
        return "lever"
    if "ashbyhq.com" in host:
        return "ashby"
    return "unknown"
